### DISCUSSION TITLE
Combining react and react-native dependency updates

### DISCUSSION THREAD

**Discussion by mschoettle** [renovatebot/renovate#20383] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.134.1

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have a React Native project and currently receive one PR for react monorepo and one for react-native (react-native + types).

The pipeline with `npm ci` fails for each of those PRs because react-native depends on a specific version of react.

I thought it might be best to combine those two in one so that this is resolved. 

### What you tried so far.

I tried to extend the existing presets for react and include the ones for react-native (taking inspiration from the `packages:react` preset):

```json
"packageRules": [
    {
      "extends": ["monorepo:react", "packages:react"],
      "matchPackageNames": [
        "@types/react-native"
      ],
      "matchPackagePrefixes": [
        "react-native"
      ]
    }
  ],
```

However, this seems to not work as there are still two separate PRs.

What's the best way to achieve this?

### Relevant debug logs

<details><summary>Logs</summary>

```
Available if necessary
```

</details>

**Comment by secustor** [comment]
Multiple `match` declarations are interpreted as `AND` therefore there are no matches for you.

```json
"packageRules": [
    {
      "extends": ["monorepo:react", "packages:react"],
      "matchPackagePrefixes": [
        "react-native",
        "@types/react-native"
      ]
    }
  ],
```
Should work for you. I have not checked how react native names their packages

**Reply by mschoettle** [reply]
That's not the behaviour I experience. It seems to be `OR` to me. For example, for `matchPackagePrefixes` and `matchPackageNames` the docs on `packageRules` states

> However you can mix together both matchPackageNames and matchPackagePatterns in the same package rule and the rule will be applied if either match.

**Comment by viceice** [comment]
that rule is missing an action, it only matched packages and nothing else. add `groupName` to group

**Reply by mschoettle** [reply]
Thanks. Without the `groupName` it indeed ignores that rule. That was not obvious to me.

**Reply by viceice** [reply]
you should have a warning in logs about the missing action

**Reply by mschoettle** [reply]
I don't see any warning in the logs about a missing action. ~~I tried to post the log but it is too long.~~ Below is the log file.

[debug.log](https://github.com/renovatebot/renovate/files/10747563/debug.log)

**Comment by mschoettle** [comment, marked as answer]
Adding this as a separate reply:

I realized that `monorepo:react` uses `matchSourceUrls` so the `react-native` packages were still not included. The following works:

```json
"packageRules": [
  {
      "extends": ["monorepo:react", "packages:react"],
      "matchSourceUrls": [
        "https://github.com/facebook/react-native",
      ],
      "matchPackageNames": [
        "@types/react-native",
      ],
      "matchPackagePrefixes": [
        "react-native",
      ],
      "groupName": "react & react-native packages",
  }
]
```
